instance/tasks.py

```diff
@@ -145,37 +169,34 @@ def spawn_appserver(
     )
 
     if appserver and mark_active_on_success:
-        make_appserver_active(appserver, active=True, deactivate_others=deactivate_old_appservers)
-
-    # Huey doesn't seem to resolve properly if the result of a task is so return False
-    return appserver or False
+        pipeline = make_appserver_active.s(appserver, active=True).then(
+            check_deactivation,
+            instance_ref_id=instance_ref_id,
+            deployment_id=deployment_id,
+            old_server_ids=old_server_ids,
+            target_count=target_count,
+        )
+        HUEY.enqueue(pipeline)
 
 
 @db_task()
-def make_appserver_active(appserver_id, active=True, deactivate_others=False):
+def make_appserver_active(appserver_id, active=True):
     """
     Mark an AppServer as active or inactive.
 
     :returns
         True if the appserver was made active,
         False if the corresponding server was not healthy and thus the appserver was not made active.
     """
-    appserver = OpenEdXAppServer.objects.get(pk=appserver_id)
+    appserver = OpenEdXAppServer.objects.get(id=appserver_id)
     if not appserver.server.status.is_healthy_state:
         appserver.server.update_status()
         if not appserver.server.status.is_healthy_state:
             logger.info('Not %s %s: ID=%s since the server is not Ready.', "Activating" if active else "Deactivating",
                         appserver, appserver_id)
             return False
-
```

Comment:
@fox, nit: I'd recommend leaving this newline intact as it improves readability.